app/src/components/features/rules/RuleBuilder/Help/index.tsx

```diff
@@ -0,0 +1,473 @@
+import "react-medium-image-zoom/dist/styles.css";
+import { RQCollapse } from "lib/design-system/components/RQCollapse";
+import { ReactComponent as Cross } from "assets/icons/cross.svg";
+import { ReactComponent as LeftArrow } from "assets/icons/left-arrow.svg";
+import { ReactComponent as RightArrow } from "assets/icons/right-arrow.svg";
+import { RuleType } from "types/rules";
+import {
+  trackDocsSidebarPrimaryCategoryClicked,
+  trackDocsSidebarSecondaryCategoryClicked,
+  trackDocsSidebarDemovideoWatched,
+  trackDocsSidebarContactUsClicked,
+} from "modules/analytics/events/common/rules";
+import "./Help.css";
+
+type ExternalLink = { title: string; link: string };
+
+const externalLinks: ExternalLink[] = [
+  {
+    title: "Tutorial videos",
+    link: APP_CONSTANTS.LINKS.YOUTUBE_TUTORIALS,
+  },
+  {
+    title: "Troubleshooting",
+    link: APP_CONSTANTS.LINKS.TUTORIALS.REDIRECT_RULE,
+  },
+  {
+    title: "Shared workspace",
+    link: APP_CONSTANTS.LINKS.DEMO_VIDEOS.TEAM_WORKSPACES,
+  },
+  {
+    title: "HTTP modifications",
+    link: APP_CONSTANTS.LINKS.REQUESTLY_DOCS_HTTP_MODIFICATIONS,
+  },
+  {
+    title: "Request a feature",
+    link: APP_CONSTANTS.LINKS.REQUESTLY_GITHUB_ISSUES,
+  },
+  {
+    title: "Ask GitHub community",
+    link: APP_CONSTANTS.LINKS.REQUESTLY_GITHUB_ISSUES, // TODO: update link
+  },
+];
+
+const Link: React.FC<{ text: string; href: string }> = ({ text, href }) => (
+  <a target="_blank" rel="noreferrer" href={href}>
+    {text}
+  </a>
+);
+
+type DocumentationListItem = {
+  title: string;
+  onClick: () => void;
+};
+
+interface HelpProps {
+  ruleType: RuleType;
+  setShowDocs: (showDocs: boolean) => void;
+}
+
+const Help: React.FC<HelpProps> = ({ ruleType, setShowDocs }) => {
+  const [isDocsVisible, setIsDocsVisible] = useState<boolean>(false);
+  const documentationListRef = useRef<HTMLDivElement | null>(null);
+  const introductionRef = useRef<HTMLDivElement | null>(null);
+  const demoVideoRef = useRef<HTMLDivElement | null>(null);
+  const howToCreateRuleRef = useRef<HTMLDivElement | null>(null);
+  const useCasesRef = useRef<HTMLDivElement | null>(null);
+  const testingRuleRef = useRef<HTMLDivElement | null>(null);
+  const examplesRef = useRef<HTMLDivElement | null>(null);
+  const faqsRef = useRef<HTMLDivElement | null>(null);
+
+  const handleDocumentationList = useCallback(
+    (ref: React.MutableRefObject<HTMLDivElement>) => {
+      if (ref.current) {
+        const target = ref.current;
+        const offsetTop = target.offsetTop;
+        (target.parentNode as HTMLElement).scrollTop = offsetTop - 87;
+      }
+    },
+    []
+  );
+
+  const documentationList: DocumentationListItem[] = useMemo(() => {
+    return [
+      {
+        title: "Introduction",
+        onClick: () => handleDocumentationList(introductionRef),
+      },
+      {
+        title: "Demo",
+        onClick: () => handleDocumentationList(demoVideoRef),
+      },
```

Comment:
In future, I would like to see it designed in a more structural way.
For example,
Each rule type exports this kind of array:
```
const redirectRuleDocSteps = [
  {
    title: "Demo",
    content: (
      <YouTubePlayer
                  width="260"
                  height="160"
                  handleOnPlay={handleDemoVideoPlay}
                  src="https://www.youtube.com/embed/lOt1kGKTq-w"
                />
    ),
    className: "rule-editor-docs-demo-video",
    analyticsEvent: "docs-demo-video",
  },
  ...
]
```
And there is a common logic which handles rendering of steps on docs index, loading content on click, triggering analytics event, etc.

Reply:
Okay.